samples/helm-deploy/templates/service.yaml

```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: carbon-aware-sdk
+  namespace: {{ .Values.namespace }}
+  labels:
+    {{- include "carbon-aware-sdk.labels" . | nindent 4 }}
+spec:
+  type: {{ .Values.service.type }}
+  ports:
+    - port: {{ .Values.service.port }}
+      targetPort: api-server-port
```

Comment:
I guess you want to specify the port what the user want to expose in values.yaml, and also it should be so. Thus `api-server-port` should be set to `port`, and also `.Values.service,.port` should be set to `targetPort`?